python/cudf/cudf/core/dataframe.py

```diff
@@ -1454,6 +1454,90 @@ def add(self, other, axis="columns", level=None, fill_value=None):
+        """
+        Modify a DataFrame in place using non-NA values from another DataFrame.
+
+        Aligns on indices. There is no return value.
+
+        Parameters
+        ----------
+
+        other : DataFrame, or object coercible into a DataFrame
+            Should have at least one matching index/column label with the 
+            original DataFrame.If a Series is passed, its name attribute must
+            be set, and that will be used as the column name to align with the 
+            original DataFrame.
+
+        join : {‘left’}, default ‘left’
+            Only left join is implemented, keeping the index and 
+            columns of the original object.
+
+        overwrite : {True, False}, default True
+            How to handle non-NA values for overlapping keys:
+            True: overwrite original DataFrame’s values with values from other.
+            False: only update values that are NA in the original DataFrame.
+
+        filter_func : callable(1d-array) -> bool 1d-array, optional
+            Can choose to replace values other than NA. 
+            Return True for values that should be updated.
+
+        errors : {‘raise’, ‘ignore’}, default ‘ignore’
+            If ‘raise’, will raise a ValueError if the DataFrame and other both contain non-NA data in the same place.
+
+
+        Returns
+        -------
+        None : method directly changes calling object
+          
+        Raises
+        -------
+        ValueError
+            - When ``errors``= 'raise' and there’s overlapping non-NA data.
+            - When ``errors`` is not either 'ignore' or 'raise'
+
+        NotImplementedError
+            - If ``join`` != ‘left’
+
+        """
+        # TODO: Support other joins
+        if join != "left": 
+            raise NotImplementedError("Only left join is supported")
+        if errors not in ["ignore", "raise"]:
+            raise ValueError("The parameter errors must be either 'ignore' or 'raise'")
+        if filter_func != None:
+            raise NotImplementedError("filter_func is not supported yet")
+
+        if not isinstance(other, DataFrame):
+            other = DataFrame(other)
+
+        other = other.reindex(self.index, axis=0)
+        other = other.reindex(self.columns, axis=1)
+
+        for col in self.columns: 
+            this = self[col]
+            that = other[col]  
+            
+            if errors == "raise":
+                mask_this = that.notna()
+                mask_that = this.notna()
+                if ((mask_this & mask_that).any()):
+                    raise ValueError("Data overlaps.")
+
+            if overwrite:
+                mask = that.isna()
+            else:
+                mask = this.notna()
+
+            # don't overwrite columns unnecessarily
+            if mask.all():
+                continue
+
+            self.loc[mask, col] = this[mask]
+            self.loc[~mask, col] = that[~mask]
```

Comment:
```suggestion
            self[col] = this.where(mask, that)
```

This can be changed to calling `where` API